packages/typescript-plugin/src/config-manager.ts

```diff
@@ -4,12 +4,15 @@ const configurationEventName = 'configuration-changed';
 
 export interface Configuration {
     enable: boolean;
+    /** Skip the Svelte detection and assume this is a Svelte project */
+    assumeIsSvelteProject: boolean;
 }
 
 export class ConfigManager {
     private emitter = new EventEmitter();
     private config: Configuration = {
-        enable: true
+        enable: true,
+        assumeIsSvelteProject: false
```

Comment:
@dummdidumm looks great, thanks!

Just to double check, `isConfigChanged` in this class still only reacts to `enable`, is that  alright?

Reply:
Yes that's right. I implemented this way because `assumeIsSvelteProject` is the very first thing that is checked, and everything else is just skipped else, so no point in reacting to something that can never be reacted to you.

Is that a problem for the Intellij integration because you create the TypeScript service eagerly, and therefore all plugins eagerly, and what you need instead is some kind of "sleep" mode where things are already prepared but not turned on yet in the plugin? I imagine that being tricky, I don't know how TypeScript would react if we lazily patch stuff. Probably the closest to this is the `enable` flag which turns off all the intellisense-related work when it's `false`, so it's almost like the sleep mode I described.

Reply:
On second take your current code is fine. No, we create TypeScript service when the first TS file is opened and we control which commands are sent in what order. We never had to use "configurePlugin" before actually and my initial understanding was wrong, so I was getting undefined at creation time. After a bit of investigation and a bug fix in protocol handling, now it works as expected 👍

Type checks for Svelte compiled classes work without node_modules installed. Zero-effort types are not injected but after `npm install` everything refreshes properly.

Thanks again!